receipt.go

```diff
@@ -0,0 +1,138 @@
+package veracity
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/datatrails/go-datatrails-common/cbor"
+	"github.com/datatrails/go-datatrails-merklelog/massifs"
+	"github.com/urfave/cli/v2"
+)
+
+func NewReceiptCmd() *cli.Command {
+	return &cli.Command{
+		Name:  "receipt",
+		Usage: "Generate a COSE Receipt of inclusion for any merkllog entry",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "output",
+				Aliases: []string{"o"},
+			},
+			&cli.Int64Flag{
+				Name: "mmrindex", Aliases: []string{"i"},
+			},
+			&cli.StringFlag{
+				Name:    "format",
+				Usage:   "override the output `FORMAT`. Supported formats: cbor (binary), hex, base64",
+				Aliases: []string{"f"},
+				Value:   "hex",
+				Action: func(ctx *cli.Context, v string) error {
+					if v != "cbor" && v != "base64" && v != "hex" {
+						return fmt.Errorf("Unsupported format '%s'. Use one of: cbor, hex, base64", v)
+					}
+					return nil
+				},
+			},
+		},
+		Action: func(cCtx *cli.Context) error {
+			cmd := &CmdCtx{}
+
+			var err error
+
+			// This command uses the structured logger for all optional output.
+			// Output not explicitly printed is silenced by default.
+			if err = cfgLogging(cmd, cCtx); err != nil {
+				return err
+			}
+
+			log := func(m string, args ...any) {
+				cmd.log.Infof(m, args...)
+			}
+
+			dataUrl := cCtx.String("data-url")
+
+			reader, err := cfgReader(cmd, cCtx, dataUrl == "")
+			if err != nil {
+				return err
+			}
+			tenantIdentity := cCtx.String("tenant")
+			if tenantIdentity == "" {
+				return fmt.Errorf("tenant identity is required")
+			}
+			log("verifying for tenant: %s", tenantIdentity)
+
+			var mmrIndex uint64
+			mmrIndex = cCtx.Uint64("mmrindex")
```

Comment:
This appears to overwrite a named argument based on a positional one, if both are supplied. What's the reasoning behind that?

Reply:
convenient ux. if you prefer exclusively on or the other I can remote the positional ?